src/types/streaming_server/request.rs

```diff
@@ -0,0 +1,323 @@
+                "key".into(),
+                // append the length of the options
+                // keep in mind that `None` options should always be treated as not-set
+                // i.e. should not be serialized
+                format!(
+                    "{key}{length}",
+                    key = self.response_key,
+                    length = options_object.len()
+                ),
+            ),
+            ("o".into(), options.to_string()),
+        ]
+    }
+}
+
+/// Server's `rar/stream` and `zip/stream` options of the query.
+///
+/// Format: `rar/stream?key={create_key}{options length}&o={options_json_string}`
+///
+/// Where all parameters are url encoded.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[cfg(feature = "experimental")]
+pub struct ArchiveStreamOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub file_idx: Option<u16>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub file_must_include: Vec<String>,
+}
+
+pub struct CreateTorrentBlobRequest {
+    pub server_url: Url,
+    pub torrent: Vec<u8>,
+}
+
+impl From<CreateTorrentBlobRequest> for Request<CreateTorrentBlobBody> {
+    fn from(val: CreateTorrentBlobRequest) -> Self {
+        let endpoint = val.server_url.join("/create").expect("url builder failed");
+
+        Request::post(endpoint.as_str())
+            .header(http::header::CONTENT_TYPE, "application/json")
+            .body(CreateTorrentBlobBody {
+                blob: hex::encode(val.torrent),
+            })
+            .expect("request builder failed")
+    }
+}
+#[derive(Serialize)]
+pub struct CreateTorrentBlobBody {
+    pub blob: String,
+}
+
+/// # Examples
+///
+/// Example which creates a request url with body for the server:
+/// `http://127.0.0.1:11470/df389295484b3059a4726dc6d8a57f71bb5f4c81/1?tr=https%3A%2F%2Fexample.com%2Fmy-awesome-video.mp4`
+///
+/// ```
+/// use stremio_core::types::streaming_server::{CreateTorrentRequest, CreatedTorrent};
+///
+/// let request: http::Request<CreatedTorrent> = CreateTorrentRequest {
+///     server_url: "http://127.0.0.1:11470/".parse().unwrap(),
+///     sources: vec!["https://example.com/my-awesome-video.mp4".into()],
+///     info_hash: "df389295484b3059a4726dc6d8a57f71bb5f4c81".parse().unwrap(),
+///     file_idx: 1,
+/// }.into();
+///
+/// assert_eq!("http://127.0.0.1:11470/df389295484b3059a4726dc6d8a57f71bb5f4c81/1?tr=https%3A%2F%2Fexample.com%2Fmy-awesome-video.mp4", request.uri().to_string());
+///  // TODO: assert_eq!(&(), request.body());
+/// ```
+pub struct CreateTorrentRequest {
+    pub server_url: Url,
+    pub sources: Vec<String>,
+    pub info_hash: InfoHash,
+    pub file_idx: u64,
+}
+
+impl From<CreateTorrentRequest> for Request<CreatedTorrent> {
+    fn from(val: CreateTorrentRequest) -> Self {
+        let url = {
+            let mut uri = val
+                .server_url
+                .join(&format!("{}/{}", val.info_hash, val.file_idx))
```

Comment:
This is the url for a stateless GET request, for a POST request this is the endpoint AFAIK - https://github.com/Stremio/enginefs/blob/dbac17e737cf3cc25e0649f3bfee15cd64962784/enginefs.js#L354

Reply:
I've removed this request as it was not being used at the time being.